src/xenia/kernel/xam/xam_net.cc

```diff
@@ -626,8 +626,7 @@ dword_result_t NetDll_socket(dword_t caller, dword_t af, dword_t type,
   if (XFAILED(result)) {
     socket->Release();
 
-    uint32_t error = xboxkrnl::xeRtlNtStatusToDosError(result);
-    XThread::SetLastError(error);
+    XThread::SetLastError(socket->GetLastWSAError());
```

Comment:
Not specific to the changes, but does the Xbox 360 have a common storage for both `GetLastError` and `NetDll_WSAGetLastError`? On Windows they're separate if I understand correctly — if that's the case on the console, we need to separate them on the guest as well (though I'm not sure if it has a predefined location in the thread context or just goes to `TlsAlloc` when `WSAStartup` is called — needs investigation).

Reply:
According to @gibbed, no, but I think we need a comment in the documentation of NetDll_WSAGetLastError that it's actually shared on the console.